### PR TITLE
Awsbatch 1kg 100sample

### PR DESCRIPTION
**Pull request by mhebrard**
Re-work of the workflow, tested with 100 samples from 1KG project re-call on DRAGEN

### CODE REVIEW

File: conf/local.config [hunk 1]
```diff
@@ -9,7 +9,7 @@ executor {
 singularity {
     enabled = true
     autoMounts = true
-    cacheDir = '/data/13000026/pipeline/dev/NPM-sample-qc-aux/containers'
+    cacheDir = '/home/ubuntu/NPM-sample-qc/containers'
```
Comment (mhebrard) [inline conf/local.config:12]: is it fine to have the full path hardcoded here ? might be an issue if the code is run in another context ( i.e. no /home/ubuntu )

File: tests/sample_params.aws.bam.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+sample_id: NA12878
+cram: s3://sg100k-benchmarking-data/datasets/HG001/1000genomes-dragen/NA12878.bam
```
Comment (mhebrard) [inline tests/sample_params.aws.bam.yml:2]: again GRIDS resources hardcoded in a public repo

File: conf/nextflow-docker.config [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        process.maxRetries = 2
+        // executor.pollInterval = "2 sec"
+        docker.enabled = true
+    }
+    awsbatch {
+        includeConfig 'base.config'
+        process.executor = "awsbatch"
+        process.region = "ap-southeast-1"
+        process.queue = "npm-sample-qc-spot70"
+        process.container = "046503905558.dkr.ecr.ap-southeast-1.amazonaws.com/npm-sample-qc:latest"
```
[the remark this answers is not in the document]
Comment (nicolas-bertin) [reply conf/nextflow-docker.config:60]: indeed @justinjj24 can you fix this before I acknowledge this PR. @mhebrard can you help @justinjj24 if it isnt obvious how to fix this ?

File: conf/nextflow.config [hunk 4]
```diff
@@ -1,28 +1,43 @@
...
+    // container = "justinjj24/npm-sample-qc:latest"
+
     // output
     keep_workdir = "false"
     publishdir = "${params.outdir}/results"
     infodir = "${params.outdir}/pipeline_info"
+
     // resources
-    resourcesdir = "/data/13000026/pipeline/dev/NPM-sample-qc-aux/resources"
+    resourcesdir = "s3://npm-grids/resources/NPM-sample-qc-aux/resources"
```
[the remark this answers is not in the document]
Comment (nicolas-bertin) [reply conf/nextflow.config:14]: indeed @justinjj24 can you fix this before I acknowledge this PR. @mhebrard can you help @justinjj24 if it isnt obvious how to fix this ?

File: tests/run.sh [hunk 5]
```diff
@@ -1,37 +1,19 @@
...
-    -profile ${profile} \
-    -resume \
-    -work-dir ${workdir} \
-    --outdir ${outdir} \
-    --keep_workdir
+nextflow run NPM-sample-qc/main.nf \
+	-config NPM-sample-qc/conf/nextflow-docker.config \
+	-params-file NPM-sample-qc/tests/sample_params.aws.bam.yml \
+	-resume \
+	--outdir s3://npm-grids/gnanakkan/npm-sample-qc/output/${sampleid} \
```
[the remark this answers is not in the document]
Comment (nicolas-bertin) [reply tests/run.sh:15]: indeed @justinjj24 can you fix this before I acknowledge this PR. @mhebrard can you help @justinjj24 if it isnt obvious how to fix this ?

File: tests/sample_params.aws.bam.yml (hunk 2 above)
Comment (nicolas-bertin) [reply tests/sample_params.aws.bam.yml:2]: indeed @justinjj24 can you fix this before I acknowledge this PR. @mhebrard can you help @justinjj24 if it isnt obvious how to fix this ?

**Comment by nicolas-bertin** [comment]
@justinjj24 , as @mhebrard rightfully pointed at there are several GRIDS resources (S3 & ECR uri) hardcoded which isnt a good thing given it is a public repo. Can you fix it before the PR is acknowledged and merged. 
@mhebrard can you help @justinjj24 if it isnt obvious how to fix those ?

File: conf/base.config [hunk 6]
```diff
@@ -7,22 +7,30 @@
  */
 process {
 
-    errorStrategy = 'finish'
+    //errorStrategy = 'finish'
```
[the remark this answers is not in the document]
Comment (nicolas-bertin) [reply conf/base.config:10]: yes

File: conf/local.config (hunk 1 above)
Comment (nicolas-bertin) [reply conf/local.config:12]: I dunno. I'd give it a pass for now and see if it turns out to be surfaced by someone as a github-issue

File: conf/nextflow-docker.config [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+    // container = "npm-sample-qc.simg"
+    // container = "justinjj24/npm-sample-qc:latest"
+
+    // output
+    keep_workdir = "false"
+    publishdir = "${params.outdir}/results"
+    infodir = "${params.outdir}/pipeline_info"
+
+    // resources
+    resourcesdir = "s3://npm-grids/resources/NPM-sample-qc-aux/resources"
```
[the remark this answers is not in the document]
Comment (nicolas-bertin) [reply conf/nextflow-docker.config:13]: yes we don't

File: tests/sample_params.aws.cram.yml [hunk 8]
```diff
@@ -0,0 +1,6 @@
+sample_id: NA12878-PCRF450-1_Proband_S1
+cram: s3://npm-grids/gnanakkan/npm-sample-qc/data/NA12878-PCRF450-1_Proband_S1.cram
```
[the remark this answers is not in the document]
Comment (nicolas-bertin) [reply tests/sample_params.aws.cram.yml:2]: indeed @justinjj24 can you fix this before I acknowledge this PR.

File: conf/base.config (hunk 6 above)
[the remark this answers is not in the document]
Comment (justinjj24) [reply conf/base.config:10]: removed the //errorStrategy = 'finish'
Different options of errorStrategy kept availble for user choice

File: conf/local.config (hunk 1 above)
Comment (justinjj24) [reply conf/local.config:12]: This file is not used in this branch, therefore I removed it.

File: conf/nextflow-docker.config (hunk 3 above)
[the remark this answers is not in the document]
Comment (justinjj24) [reply conf/nextflow-docker.config:60]: process.container = "xxxxxxxxx.dkr.ecr.ap-southeast-1.amazonaws.com/npm-sample-qc:latest"

File: conf/nextflow.config (hunk 4 above)
[the remark this answers is not in the document]
Comment (justinjj24) [reply conf/nextflow.config:14]: This file is not used in this branch, therefore I removed it.

File: tests/run.sh (hunk 5 above)
[the remark this answers is not in the document]
Comment (justinjj24) [reply tests/run.sh:15]: --outdir output/${sampleid} \
-bucket-dir output/${sampleid}-work \

File: tests/sample_params.aws.bam.yml (hunk 2 above)
Comment (justinjj24) [reply tests/sample_params.aws.bam.yml:2]: fixed to cram: s3://1000genomes-dragen-3.7.6/data/individuals/hg38-graph-based/NA12878/NA12878.bam

File: tests/sample_params.aws.cram.yml (hunk 8 above)
[the remark this answers is not in the document]
Comment (justinjj24) [reply tests/sample_params.aws.cram.yml:2]: fixed to cram: "" // path to your cram file eg. "s3://npm-sample-qc/data/NA12878.cram"

**Comment by mhebrard** [comment]
Re-opening to handle comments and fixes

**Review by nicolas-bertin — APPROVED** [review]

**Review by nicolas-bertin — APPROVED** [review]

**Review by nicolas-bertin — APPROVED** [review]

**Pull request merged**